pkg/agent/sysadvisor/metacache/metacache.go

```diff
@@ -46,6 +46,8 @@ const (
 
 // MetaReader provides a standard interface to refer to metadata type
 type MetaReader interface {
+	// GetContainers returns containers filtered by containerFilter
+	GetContainers(containerFilter func(*types.ContainerInfo) bool) []*types.ContainerInfo
```

Comment:
it is unnecessary to add this function since it can be totally achieved by `RangeContainer`